pkg/scheduler/framework/plugins/volumebinding/volume_binding.go

```diff
@@ -157,9 +157,10 @@ func (pl *VolumeBinding) PreBind(ctx context.Context, cs *framework.CycleState,
 	return nil
 }
 
-// Unreserve clears pod binding state.
-// TODO(#90962) Revert assumed PV/PVC cache
+// Unreserve clears assumed PV and PVC cache and pod binding state.
+// It's idempotent, and does nothing if no cache or binding state found for the given pod.
 func (pl *VolumeBinding) Unreserve(ctx context.Context, cs *framework.CycleState, pod *v1.Pod, nodeName string) {
+	pl.Binder.RevertAssumedPodVolumes(pod, nodeName)
```

Comment:
Thank you for the PR! Is it possible to add a unit test for this scenario that we're fixing?

Reply:
I've added a unit test for SchedulerVolumeBinder. I can add a unit test that checks RevertAssumedPodVolumes is called in Unreserve, but I don't think it would make much sense.